barretenberg/cpp/src/barretenberg/dsl/acir_format/block_constraint.cpp

```diff
@@ -24,6 +24,71 @@ template <typename Builder> bb::stdlib::field_t<Builder> poly_to_field_ct(const
 
 template <typename Builder>
 void create_block_constraints(Builder& builder, const BlockConstraint constraint, bool has_valid_witness_assignments)
+    requires IsNotMegaBuilder<Builder>
```

Comment:
One thing that's not clear to me here is how the databus will be handled if the backend uses Ultra. If I understand correctly, we create databus constraints regardless of arithmetization but then simply do nothing with them in the case of Ultra. This seems like it could easily lead to silent issues. I was imagining that in the case of Ultra we would somehow reinterpret the databus functionality using RAM/ROM. What do you think is the right way to handle this?

Reply:
Yes it's handled like a ROM:
```
 switch (constraint.type) {
    case BlockType::CallData:
    case BlockType::ReturnData:
    case BlockType::ROM: {...
```